src/components/update-service/update-service.controller.ts

```diff
@@ -213,74 +213,28 @@ export class UpdateServiceController implements angular.IController {
     this.ctrl.orderComplete = false;
     this.ctrl.error = null;
 
-    var parameterData: any = this.getParameters(this.ctrl.parameterData);
+    let parameterData: any = this.getParameters(this.ctrl.parameterData);
 
     // Update the instance parameters
-    var originalParameters: any = _.get(this.ctrl.serviceInstance, 'spec.parameters');
-    var originalKeys: any = _.map(originalParameters, (value: any, key: string) => {
-      // Make sure properties with a '.' in the name aren't treated as paths.
+    let originalParameters: object = _.get(this.ctrl.serviceInstance, 'spec.parameters');
+    // NOTE: should be string[][], but compiler doesn't handle this
+    let originalKeys: any[] = _.map(originalParameters, (value: any, key: string) => {
```

Comment:
Does this work?

```ts
const originalKeys: string[][] = _.map(originalParameters, (value: any, key: string): string[] => {
```

I'm not sure why we need a multidimensional array here, though.

Reply:
It's a copy-paste move, the code was originally a multidimensional array.  Not sure why, really.  I was aiming to type it anyway, though.